clientconn.go

```diff
@@ -153,6 +153,18 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 			})
 		}
 		cc.csMgr.channelzID = cc.channelzID
+		defer func() {
```

Comment:
cc.Close() will also remove channel and add trace channel deleted.

Move `RegisterChannel` down (right before defer with `cc.Close()`), then this should be covered by cc.Close().